File: problem/Max_K-Sum_Pairs/maxOperations.py
class Solution:
    def maxOperations(self, nums, k):
        n = len(nums)
        i = 0
        j = n - 1
        count = 0
        nums.sort()
        while i < j:
            sum = nums[i] + nums[j]
            if sum == k:
                count += 1
                i += 1
                j -= 1
            if sum < k:
                i += 1
            if sum > k:
                j -= 1
        return count

File: problem/Max_K-Sum_Pairs/maxOperations.java
class Solution {
    public int maxOperations(int[] nums, int k) {
        int n = nums.length;
        int i = 0;
        int j = n - 1;
        int count = 0;
        Arrays.sort(nums);
        while (i < j) {
            int sum = nums[i] + nums[j];
            if (sum == k) {
                count++;
                i++;
                j--;
            }
            if (sum < k) {
                i++;
            }
            if (sum > k) {
                j--;
            }
        }
        return count;
    }
}
